tests/basic/distribute/directory-rebalance.t

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+. $(dirname $0)/../../include.rc
+. $(dirname $0)/../../volume.rc
+
+#This tests checks if directory rebalance works as expected
+TESTS_EXPECTED_IN_LOOP=40
+
+cleanup;
+
+#https://stackoverflow.com/questions/5799303/print-a-character-repeatedly-in-bash/17030976
+#'a' repeated 255 times to test that the index/log files are generated correctly
+DIR1=$(printf "%0.sa" $(seq 1 255))
+
+TEST glusterd
+TEST pidof glusterd
+TEST $CLI volume create $V0 $H0:$B0/${V0}0
+TEST $CLI volume set $V0 performance.quick-read off
+TEST $CLI volume set $V0 performance.io-cache off
+TEST $CLI volume set $V0 performance.write-behind off
+TEST $CLI volume set $V0 performance.stat-prefetch off
+TEST $CLI volume set $V0 performance.read-ahead off
+TEST $CLI volume start $V0
+TEST $GFS --volfile-id=/$V0 --volfile-server=$H0 $M0
+TEST mkdir $M0/${DIR1}
+for i in {1..10}
+do
+    TEST_IN_LOOP dd if=/dev/zero of=$M0/$i bs=1k count=1
+    TEST_IN_LOOP dd if=/dev/zero of=$M0/${DIR1}/$i bs=1k count=1
+    TEST_IN_LOOP mknod $M0/block-${i} b 4 5
+    TEST_IN_LOOP "touch $M0/${i}\*\)\(\'" #Files with special chars in them to check that the index is built correctly and files are migrated correctly
+done
+TEST $CLI volume add-brick $V0 $H0:$B0/${V0}1 force
+TEST $CLI volume rebalance $V0 fix-layout start
+EXPECT_WITHIN $REBALANCE_TIMEOUT "fix-layout completed" fix-layout_status_field $V0
+TEST $(dirname $0)/../../../extras/rebalance/directory-rebalance.py $M0/${DIR1}
+
+root_file_success_count=0
+root_file_failure_count=0
+dir_file_success_count=0
+dir_file_failure_count=0
+for i in {1..10}
+do
+    #root directory files shouldn't be migrated
+    if [ -f $B0/${V0}0/$i ]
+    then
+        root_file_success_count=$((root_file_success_count+1))
+    fi
+
+    #Some files in ${DIR1} should be migrated
+    if [ -f $B0/${V0}0/${DIR1}/$i ]
+    then
+        dir_file_success_count=$((dir_file_success_count+1))
+    else
+        dir_file_failure_count=$((dir_file_failure_count+1))
```

Comment:
I would have preferred dir_file_migrated_count and dir_file_not_migrated_count
and increase it in respective "then" "else" and check accordingly.
dir_file_failure_count looks like rebalance have failed.. it is just for better understanding while reading code.

Reply:
I will use dir_file_scanned_count and dir_file_migrated_count

Reply:
Done